src/Compilers/VisualBasic/Portable/Semantics/OverloadResolution.vb

```diff
@@ -1187,6 +1259,248 @@ ResolutionComplete:
         End Function
 
+        Private Shared ReadOnly s_poolInstance As ObjectPool(Of PooledDictionary(Of NamedTypeSymbol, NamedTypeSymbol)) =
+            PooledDictionary(Of NamedTypeSymbol, NamedTypeSymbol).CreatePool(SymbolEqualityComparer.IgnoreAll)
+
+        Private Shared Sub RemoveLowerPriorityMembers(
+            candidates As ArrayBuilder(Of CandidateAnalysisResult),
+            ByRef applicableCandidates As Integer,
+            ByRef applicableNarrowingCandidates As Integer
+        )
+            ' PROTOTYPE(Priority): Check feature support
+
+            Dim candidateInfoByDeclaringType = PooledDictionary(Of NamedTypeSymbol, OneOrMany(Of (Index As Integer, Priority As Integer))).GetInstance()
+
+            For i As Integer = 0 To candidates.Count - 1 Step 1
+
+                Dim current As CandidateAnalysisResult = candidates(i)
+
+                If current.State <> CandidateAnalysisResultState.Applicable Then
+                    Continue For
+                End If
+
+                Dim priorityInfo As (Source As NamedTypeSymbol, Priority As Integer) = current.Candidate.GetOverloadResolutionPriorityInfo()
+
+                Dim siblings As OneOrMany(Of (Index As Integer, Priority As Integer)) = Nothing
+
+                If candidateInfoByDeclaringType.TryGetValue(priorityInfo.Source, siblings) Then
+                    candidateInfoByDeclaringType(priorityInfo.Source) = siblings.Add((i, priorityInfo.Priority))
+                Else
+                    candidateInfoByDeclaringType.Add(priorityInfo.Source, OneOrMany.Create((i, priorityInfo.Priority)))
+                End If
+            Next
+
+            For Each siblings In candidateInfoByDeclaringType.Values
+                If siblings.Count = 1 Then
+                    Continue For
+                End If
+
+                Dim maxPriority = Integer.MinValue
+
+                For Each info In siblings
+                    If maxPriority < info.Priority Then
+                        maxPriority = info.Priority
+                    End If
+                Next
+
+                For Each info In siblings
+                    If maxPriority <> info.Priority Then
+                        Dim toDeprioritize = candidates(info.Index)
+                        applicableCandidates -= 1
+                        If toDeprioritize.RequiresNarrowingConversion Then
+                            applicableNarrowingCandidates -= 1
+                        End If
+                        toDeprioritize.State = CandidateAnalysisResultState.LessApplicable
+                        candidates(info.Index) = toDeprioritize
+                    End If
+                Next
+            Next
+
+            candidateInfoByDeclaringType.Free()
+        End Sub
+
+        Private Shared Sub ApplyTieBreakingRulesSkippedByCombineCandidates(
```

Comment:
>ApplyTieBreakingRulesSkippedByCombineCandidates [](http://example.com/codeflow?start=27&length=47)

Effect of changes to `CombineCandidates` with this fixup was tested across all unit-tests that we have right now - [https://github.com/dotnet/roslyn/pull/75792](https://github.com/dotnet/roslyn/pull/75792).